scripts/policy/frameworks/cluster/controller/main.zeek

```diff
@@ -141,57 +199,89 @@ event ClusterController::API::set_configuration_request(reqid: string, config: C
 	{
 	ClusterController::Log::info(fmt("rx ClusterController::API::set_configuration_request %s", reqid));
 
+	local res: ClusterController::Types::Result;
 	local req = ClusterController::Request::create(reqid);
-	req$set_configuration_state = ClusterController::Request::SetConfigurationState();
+	req$set_configuration_state = ClusterController::Request::SetConfigurationState($config = config);
 
-	# Compare new configuration to the current one and send updates
-	# to the instances as needed.
-	if ( config?$instances )
+	# At the moment there can only be one pending request.
```

Comment:
It'd certainly be better. Right now the idea is simply that if you submit a request while the controller hasn't fully processed the former, you're simply told that fact and rejected. It's another foundation building block because I'm sure such request queueing will come in handy for other types of requests.